pallets/stream/src/lib.rs

```diff
@@ -135,6 +140,20 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	/// stream uniques stored on chain.
+	/// It maps from a stream identifier and hash to its details.
+	#[pallet::storage]
+	#[pallet::getter(fn attestations)]
+	pub type Attestations<T> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		StreamIdOf,
+		Blake2_128Concat,
+		StreamDigestOf<T>,
+		StreamAttestationEntryOf<T>,
```

Comment:
This can be StreamEnty (or rename it to AttestationDetails). With that, We can change Streams to StreamIdentifier, which maps an identifier to the latest digest.

Reply:
updated this. 

Other storage update IMO can come in as separate PR. So, that review will be cleaner.